cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -429,9 +429,25 @@ runTxBuildCmd
               let consensusMode = consensusModeOnly cModeParams
               case consensusMode of
                 CardanoMode -> do
-                  (nodeEraUTxO, _, eraHistory, systemStart, _) <-
-                    queryStateForBalancedTx_ nodeEra nid allTxInputs
-                      & handleQueryConvenienceErrors_
+                  (nodeEraUTxO, eraHistory, systemStart) <-
+                    (do
+                      let cModeParams2 = CardanoModeParams $ EpochSlots 21600
+
+                      let localNodeConnInfo2 = LocalNodeConnectInfo cModeParams2 nid sockPath
+
+                      qSbe <- getSbe (cardanoEraStyle nodeEra) & OO.hoistEither
+
+                      qeInMode <- toEraInMode nodeEra CardanoMode
+                        & OO.hoistMaybe (EraConsensusModeMismatch (AnyConsensusMode CardanoMode) (getIsCardanoEraConstraint nodeEra $ AnyCardanoEra nodeEra))
+
+                      -- Query execution
+                      executeLocalStateQueryExpr_ localNodeConnInfo2 Nothing $ do
+                        utxo <- queryUtxo_ qeInMode qSbe allTxInputs
+                        eraHistory <- queryEraHistory_
+                        systemStart <- querySystemStart_
+
+                        pure (utxo, eraHistory, systemStart)
+                    ) & handleQueryConvenienceErrors_
```

Comment:
Inline `queryStateForBalancedTx_` to avoid unnecessarily querying for `pparams` and `stakePools` which are never used.